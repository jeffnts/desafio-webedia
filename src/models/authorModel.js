import mongoose from 'mongoose'
import mongoosePaginate from 'mongoose-paginate-v2'

const Schema = mongoose.Schema

const authorSchema = new Schema({
    name:{
        type: String,
        required: true
    }
    
})

authorSchema.plugin(mongoosePaginate)

export default  mongoose.model('Author', authorSchema)