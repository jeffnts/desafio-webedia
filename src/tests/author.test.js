const app = require('../app')
const request = require('supertest')(app)
import authorModel from '../models/authorModel'
const connection = require('../config/database/testDB')
const {sanitizeTestObject} = require('../config/tests/sanitizeTestObject')

const dropCollection = async () => authorModel.remove()

beforeAll(() => {
    dropCollection()    
})
    
afterAll(() => {
connection.disconect()
})

let author = {
  "_id": "TESTOBJECT01",
  "name": "Monteiro Lobato"
}


describe('Author tests', () =>{
  describe('POST method', () =>{
    test('It should create a new author.', async () =>{
      const res = await request.post('/api/authors').send(author)
      expect(sanitizeTestObject(res.body)).toMatchSnapshot()
      expect(res.statusCode).toBe(201)
    })
  })

  describe('GET method', () =>{
    test('It shoul return all authors', async () =>{
      const res = await request.get('/api/authors')
      expect(sanitizeTestObject(res.body)).toMatchSnapshot()
      expect(res.statusCode).toBe(200)
    })

    test('It should return the authors according the id passed as parameter', async () =>{
      const res = await request.get(`/api/authors/${author._id}`)
      expect(sanitizeTestObject(res.body)).toMatchSnapshot()
      expect(res.statusCode).toBe(200)
    })
    test('It should not return the authors with the wrong the id passed as parameter', async () =>{
      const res = await request.get('/api/authors/TESTOBJECT02')
      expect(sanitizeTestObject(res.body)).toMatchSnapshot()
      expect(res.statusCode).toBe(404)
    })
  })

  describe('PUT method', ()=>{
    test('It should update the author according the id passed as parameter', async () =>{
      const res = await request.put(`/api/authors/${author._id}`).send({name: 'Zé do Livro'})
      expect(sanitizeTestObject(res.body)).toMatchSnapshot()
      expect(res.statusCode).toBe(200)
    })
    test('It should not update the author with the wrong id passed as parameter', async () =>{
      const res = await request.put('/api/authors/TESTOBJECT02').send({name: 'Zé do Livro'})
      expect(sanitizeTestObject(res.body)).toMatchSnapshot()
      expect(res.statusCode).toBe(404)
    })
  })

  describe('DELETE method', ()=>{
    test('It should not delete the author with the wrong id passed as parameter', async () =>{
      const res = await request.delete('/api/authors/TESTOBJECT02')
      expect(sanitizeTestObject(res.body)).toMatchSnapshot()
      expect(res.statusCode).toBe(404)
    })
    test('It should delete the author according the id passed as parameter', async () =>{
      const res = await request.delete(`/api/authors/${author._id}`)
      expect(sanitizeTestObject(res.body)).toMatchSnapshot()
      expect(res.statusCode).toBe(200)
    })

  })
})