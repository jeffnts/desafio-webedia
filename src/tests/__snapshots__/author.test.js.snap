// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author tests DELETE method It should delete the author according the id passed as parameter 1`] = `
Object {
  "message": "Autor removido com sucesso!",
}
`;

exports[`Author tests DELETE method It should not delete the author with the wrong id passed as parameter 1`] = `
Object {
  "message": "Este autor não existe.",
}
`;

exports[`Author tests GET method It shoul return all authors 1`] = `
Object {
  "authors": Object {
    "docs": Array [
      Object {
        "__v": 0,
        "_id": "ObjectId",
        "name": "Monteiro Lobato",
      },
    ],
    "limit": "EMPTY",
    "offset": "EMPTY",
    "totalDocs": 1,
  },
}
`;

exports[`Author tests GET method It should return the authors according the id passed as parameter 1`] = `
Object {
  "author": Object {
    "__v": 0,
    "_id": "ObjectId",
    "name": "Monteiro Lobato",
  },
}
`;

exports[`Author tests POST method It should create a new author. 1`] = `
Object {
  "message": "Autor criado com sucesso!",
}
`;

exports[`Author tests PUT method It should update the author according the id passed as parameter 1`] = `
Object {
  "message": "Este autor não existe.",
}
`;
